standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/RawStore.java

```diff
@@ -1623,6 +1623,15 @@ void dropFunction(String catName, String dbName, String funcName)
    */
   List<String> getFunctions(String catName, String dbName, String pattern) throws MetaException;
 
+  /**
+   * Retrieve list of function names based on name pattern.
+   * @param dbName database name
+   * @param pattern pattern to match
+   * @return functions that match the pattern
+   * @throws MetaException incorrectly specified function
+   */
+  List<Function> getFunctionsInDb(String catName, String dbName, String pattern) throws MetaException;
```

Comment:
I was thinking how we can merge `List<String> getFunctions` and `List<Function> getFunctionsInDb`, can we change it to ` <T> void getFunctions(String catName, String dbName, String pattern, List<T> result) throws MetaException;` and mark the original `getFunctions` as deprecated?

Reply:
ack